src/services/validation.js

```diff
@@ -66,28 +90,51 @@ async function checkUserIdInFirestore(
  * @param {String} dataSource - contains information about where and how the data is stored.
  * @param {String} authType - the authentication method the user has used
  * @param {Number} validateCount - indicates how many times the user has been validated
+ * @param {Object} birthdate - contains the month, day and year
+ * @param {Boolean} isExtraInputValidationRequired - indicates if there are any extra fields being validated
  */
 export async function validateID(
   userID,
   dataSource,
   authType,
-  validateCount = 0
+  validateCount = 0,
+  birthdate,
+  isExtraInputValidationRequired
 ) {
   if (dataSource["type"] == "Firestore") {
-    if (authType == "ID") {
-      return checkUserIdInFirestore(
-        userID,
-        validateCount,
-        dataSource["name"],
-        dataSource["column"]
-      );
-    }
-    if (authType == "OTP") {
-      return checkPhoneNumberInFirestore(
-        userID,
-        dataSource["name"],
-        dataSource["column"]
+    if (isExtraInputValidationRequired) {
+      return (
+        checkBirthdateInFirestore(
+          birthdate,
+          userID,
+          dataSource["name"],
+          "date_of_birth"
+        ) &&
+        checkUserIdInFirestore(
+          userID,
+          validateCount,
+          dataSource["name"],
+          dataSource["column"],
+          isExtraInputValidationRequired
+        )
       );
+    } else {
+      if (authType.includes("ID")) {
+        return checkUserIdInFirestore(
```

Comment:
How come even if the `datasource` is not `Firestore`, we're still checking if the `id` is in `Firestore`?

Reply:
So, if you look at the code, there is no checking for another data source. I'll make the change.

Reply:
yes, just code prettiness and understandability change needed here

Reply:
Actually, I misunderstood your comment. This else statement is inside `dataSource[type] == 'Firestore'.